src/ModuleList/Quiz/Quiz.tsx

```diff
@@ -1,22 +1,78 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useParams } from 'react-router-dom';
+import { SimpleGrid, Box, Flex } from '@chakra-ui/core';
+import { SubmitQuiz, CheckIcon, IncorrectIcon } from './styled';
+import { useQuizMeta } from 'src/common/api/curriculum/quiz';
+import { Question } from 'src/common/api/curriculum/quiz/requests';
 
 interface Params {
-  moduleId: string;
   quizId: string;
 }
 
+interface StringMap {
+  [key: string]: boolean;
+}
+
 const Quiz = () => {
-  const { moduleId, quizId } = useParams<Params>();
+  const { quizId } = useParams<Params>();
+
+  const { data } = useQuizMeta(Number(quizId));
+
+  const [questionIndexesAnsweredCorrectly, setQuestionIndexesAnsweredCorrectly] = useState<StringMap>({});
+  const [quizIsSubmitted, setQuizIsSubmitted] = useState<boolean>(false);
+
+  const handleAnsweredQuestion = (q: Question, qIndex: number, optionIndex: number) => {
+    return (e: React.ChangeEvent<any>) => {
+      questionIndexesAnsweredCorrectly[qIndex] = q.answer === optionIndex;
+      setQuestionIndexesAnsweredCorrectly(questionIndexesAnsweredCorrectly);
+    };
+  };
+
+  const checkQuiz = () => {
+    setQuizIsSubmitted(true);
+  };
+
+  const getLabelStyle = (q: Question, qIndex: number, optionIndex: number) => {
+    let color = '#FFFFFF';
+    if (quizIsSubmitted &&
+      !!questionIndexesAnsweredCorrectly[qIndex] === false &&
+      q.answer === optionIndex) {
+      color = '#C5A3B4';
+    }
+    return { color };
+  };
 
   return (
     <div>
-      <h1>
-        Module: {moduleId}
-      </h1>
-      <h2>
-        Quiz: {quizId}
-      </h2>
+      <SimpleGrid columns={1} spacing={10}>
+        {Array.isArray(data) && data.map((q: Question, qIndex: number) => (
+          <Box p={5} color="white" key={qIndex}>
+            <Flex>
+              {quizIsSubmitted && (questionIndexesAnsweredCorrectly[qIndex] ?
+                <CheckIcon color="green.500"/> : <IncorrectIcon color="red.500" />)}
+              <p>{ `${qIndex + 1}. ${q.question}`}</p>
+            </Flex>
+            <SimpleGrid columns={2} spacingX={1}>
+              {q.options.map((option, optionIndex) => (
+                <div key={optionIndex}>
```

Comment:
We probably shouldn't use keys as indexes. This is because of how React does diffing to check and see whether or not to re-render an item. Since we're not adding anything to these arrays and they wont change it doesn't really matter too much, however, it's probably not good practice so maybe try to use some string or ID that wont change? Possibly just using the `option` variable...

See [this](https://medium.com/better-programming/why-react-keys-matter-an-introduction-136b7447cefc) article for a better explanation.